### PR TITLE
feat: generalize dual Python IPC and speed-up conda builds

### PR DESCRIPTION
**Pull request by isidentical**
More description about the actual algoritm is under `DualPythonIPC` class.

### CODE REVIEW

File: src/fal/packages/environments/base.py [hunk 1]
```diff
@@ -176,3 +178,71 @@ def poll_until_result(
...
+            env={"PYTHONPATH": python_path, **os.environ},
+        )
+
+
+def python_path_for(*search_paths) -> str:
+    assert len(search_paths) >= 1
+    return os.pathsep.join(
+        # sysconfig takes the virtual environment path and returns
+        # the directory where all the site packages are located.
+        sysconfig.get_path("purelib", vars={"base": search_path})
```
Comment (chamini2) [inline src/fal/packages/environments/base.py:234]: ATL: what is purelib?

File: src/fal/packages/environments/conda.py [hunk 2]
```diff
@@ -67,48 +65,18 @@ def _get_or_create(self) -> Path:
...
-                python_executable,
-                isolated_runner.__file__,
-                bridge.encode_service_address(service.address),
-            ],
-        )
+    def open_connection(self, conn_info: Path) -> DualPythonIPC:
+        primary_env = get_primary_virtual_env()
+        primary_env_path = primary_env.get_or_create()
+        secondary_env = conn_info
+        return DualPythonIPC(self, primary_env_path, secondary_env)
```
Comment (chamini2) [inline src/fal/packages/environments/conda.py:79]:
```suggestion
        secondary_env_path = conn_info
        return DualPythonIPC(self, primary_env_path, secondary_env_path)
```

**Review by chamini2 — APPROVED** [review]

File: src/fal/packages/environments/conda.py (hunk 2 above)
Comment (isidentical) [reply src/fal/packages/environments/conda.py:79]: thx!

File: src/fal/packages/environments/base.py (hunk 1 above)
Comment (isidentical) [reply src/fal/packages/environments/base.py:234]: `purelib` points out to the path where the actual packages are located. E.g. my venv is under `~/.cache/fal/xxx` but the packages actually reside in `~/.cache/fal/xxx/lib/site-packages/something` (depending on the platform), using `purelib` with the base of our search_path (which is the venv directory, or conda) we can find the proper path on where all the site-packages are located.

Comment (isidentical) [reply src/fal/packages/environments/base.py:234]: Definitions are encoded in here (and packaging tools like pip/conda use them) https://github.com/python/cpython/blob/c112489f877a37cd6e887f0893fc1f4843c168fa/Lib/sysconfig.py#L88-L118

**Pull request merged**